ansible/roles/edu-sharing/templates/edu-sharing.conf.j2

```diff
@@ -16,6 +16,42 @@ repository: {
 {% endif %}
 }
 
+// Default connectors for the repository will be none
+connectorList:{
+    connectors:[]
+}
+
+
+{% if enable_edu_connector is defined and enable_edu_connector and edu_connector_enable_modules.tiny_mce %}
+// enable TINYMCE
+connectorList.connectors +=  {
+   id:"TINYMCE", icon:"edit", showNew: true, onlyDesktop: true, hasViewMode: true,
+   filetypes:[
+     {mimetype: "application/text",filetype: "html", ccressourcetype: "html", createable: true,editable: true}
+   ]
+}
+{% endif %}
+{% if enable_edu_connector is defined and enable_edu_connector and edu_connector_enable_modules.h5p %}
+// enable H5P
+connectorList.connectors +=  {
+   id:"H5P", icon:"edit", showNew: true, onlyDesktop: true, hasViewMode: true,
+   filetypes:[
+     {mimetype: "application/zip",filetype: "h5p", ccressourcetype: "h5p", createable: true,editable: true}
+   ]
+}
+{% endif %}
+
+{% if enable_only_office is defined and enable_only_office and edu_connector_enable_modules.only_office %}
+// enable ONLY_OFFICE
+connectorList.connectors +=  {
+   id:"ONLY_OFFICE", icon:"edit", showNew: true, onlyDesktop: true, hasViewMode: true,
+   filetypes:[
+     {mimetype: "application/vnd.openxmlformats-officedocument.wordprocessingml.document",filetype: "docx",createable: true,editable: true},
+     {mimetype: "application/vnd.openxmlformats-officedocument.presentationml.presentation",filetype: "pptx",createable: true,editable: true},
+     {mimetype: "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",filetype: "xlsx",createable: true,editable: true},
+   ]
+{% endif %}
+
```

Comment:
Can this be generalized? Currently each type (and each type that will be available in the future) must always be implemented separately. For maintainability, it would be better if the implementation is universal for all types and the individual values are only stored in the configuration. In this way, only the configuration would have to be adapted in case of changes instead of new implementation + installation of a new version.

Reply:
Hello @mirjan-hoffmann 

I have implemented an generic way